### ISSUE TITLE
[CoE Starter Kit - BUG] Data not updating after host name change

### ISSUE THREAD

**Issue by Dom-Hen** [microsoft/coe-starter-kit#8661]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

CoE was configured by a colleague that has left. After picking this up I noticed that there are no cloud flows (apps and others as well) created recently (by any user in any environment). Chasing the issue found that some CoE flows are failing with the error finding the host (name was changed after the CoE was deployed). 
The connection references in solution look fine (at least it seems it works some times) but have not tried creating new ones due to the issue with unmanaged layers on the solution. I have turned off and on the flow as the guide says but still no joy.
![CoE error](https://github.com/user-attachments/assets/7a98cc15-93f6-4d6e-8ccc-b62ee5c41b3c)

### Expected Behavior

Created a flow on a monitored environment but is not coming/showing in the CoE.
![CoE expected behaviour](https://github.com/user-attachments/assets/f7d619cc-3110-4fed-b142-10507f9b412e)

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.29

### What app or flow are you having the issue with?

Admin | Sync Template v4 (Flows)

### What method are you using to get inventory and telemetry?

Cloud flows

AB#3816

**Comment by Jenefer-Monroe** [comment]
Unmanaged layers will make it difficult to debug as it means you are not in the same state as we shipped with. 
But we can give it a try. 

First of all, you'll likely want to upgrade soon. There was a product change that caused us to have to upgrade the creator kit version and the apps to use it. So you'll want to upgrade to at least Creator Kit Core 1.0.20240529.1 and to at least the June release of Core (4.32.2) (I'll post more below)

And please be sure that each time you upgrade you boot the setup wizard as it will make sure that all new flows and env vars are filled. (I'll post more below)

Upgrade will also reset all the flows in the kit (unless there are unmanaged layers). And typically things like cached host names are cleared from flows with upgrade. 

Lastly, you'll also want to note that issues connecting with other environments may be due to a product change around the way the product treats escalated roles (PPAdmin Role). There is a workaround thats needed. (I'll post more below)

So in short, my suggestion is:
1. Try to understand the unmanaged layers created so that they can be removed or re-applied
2. Upgrade Creator Kit Core
3. Upgrade Core (and gov/nurture if you use them)
4. Run the setup wizard
5. Implement the workaround for the escalated roles change (see below)
6. See what state you are in

**Comment by Jenefer-Monroe** [comment]
### Use the setup wizard
Its highly recommended that you use the Setup Wizard to walk you through setup and upgrades. It will ensure all the new flows are turned on in the correct order, trigger things like population of the bookmarks screen and do other cleanup steps needed.
You should do this both for upgrades and clean installs.
To use it, first perform the import of the solution install or upgrade manually, then boot the app and walk through the screens.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/b84cc1ff-383a-497c-9236-569e8313c73c)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/ed02f0d0-c276-48cc-90af-47e5bf5a928e)

**Comment by Dom-Hen** [comment]
Hello Jenefer!
Thank you for the quick reply. I will try your approach as I thought about a fresh start for it but if the upgrade works, it will be better!

**Comment by Jenefer-Monroe** [comment]
We are here to help so please do let me know how I can assist in your ramp up!

**Comment by Dom-Hen** [comment]
Hi Jenefer! quick question: i did push the new solution in and all went well but when going through the wizard, on step "Run inventory flows" it complained that various flows are off. And they cannot be turned "on" due to the connection reference being invalid. I assumed the upgrade would refresh these connection? in order to "move on" I could reconnect each individually but it would create a solution layer on each of the flows that were not able to turn on.

For reference, before the upgrade all any/all solutions layers existing on the flows was removed.

**Comment by Dom-Hen** [comment]
All seems to be back and connected, although I had some issues with the "admin_CoECoreTeams" as it was not connected at all and i was not sure of the details.
Will try the same on our production environment

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
